### PR TITLE
8260592: jpackage tests fail when Desktop is not supported

### PR DESCRIPTION
**Pull request by shipilev**
If you run x86 configuration (cross-compiled on x86_64), then many tests would fail with:

```
$ CONF=linux-x86-server-fastdebug make images run-test TEST=tools/jpackage
...
can not load libgnomevfs-2.so
Exception in thread "main" java.lang.ExceptionInInitializerError
Caused by: java.lang.UnsupportedOperationException: Desktop API is not supported on the current platform
at java.desktop/java.awt.Desktop.getDesktop(Unknown Source)
at com.that/com.that.main.Florence.createInstance(Unknown Source)
at com.that/com.that.main.Florence.<clinit>(Unknown Source)
Failed to launch JVM
java.lang.AssertionError: Expected [0]. Actual [1]: Check command [/home/shade/trunks/jdk/build/linux-x86-server-fastdebug/test-support/jtreg_test_jdk_tools_jpackage/scratch/10/./testMainLauncherIsModular.ed4f638d/output/MainLauncherIsModularAddLauncherTest/bin/ModularAppLauncher](1) exited with 0 code
```

The tests probably need to check `Desktop.isDesktopSupported()`, similarly how they check `GraphicsEnvironment.isHeadless()`.

Additional testing:
 - [x] Linux x86_32 `tools/jpackage` (now pass)
 - [x] Linux x86_64 `tools/jpackage` (still pass)

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8260592](https://bugs.openjdk.java.net/browse/JDK-8260592): jpackage tests fail when Desktop is not supported

### Reviewers
 * [Alexey Semenyuk](https://openjdk.java.net/census#asemenyuk) (@alexeysemenyukoracle - Committer)
 * [Andy Herrick](https://openjdk.java.net/census#herrick) (@andyherrick - **Reviewer**) ⚠️ Review applies to c0975ae319b0b19a3f90a6337ddd23320bcbe334

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2291/head:pull/2291`
`$ git checkout pull/2291`

### CODE REVIEW

File: test/jdk/tools/jpackage/apps/image/Hello.java [hunk 1]
```diff
@@ -163,6 +163,12 @@ private static Hello createInstance() {
             ex.printStackTrace();
         }
 
+        if (!Desktop.isDesktopSupported()) {
```
Comment (alexeysemenyukoracle) [inline test/jdk/tools/jpackage/apps/image/Hello.java:166]: Would it make more sense to replace
`if (GraphicsEnvironment.isHeadless())`
with
`if (GraphicsEnvironment.isHeadless() || !Desktop.isDesktopSupported())`
without need to add another `trace()` call?

**Review by alexeysemenyukoracle — CHANGES_REQUESTED** [review]

**Review by andyherrick — APPROVED** [review]

File: test/jdk/tools/jpackage/apps/image/Hello.java (hunk 1 above)
Comment (shipilev) [reply test/jdk/tools/jpackage/apps/image/Hello.java:166]: I would prefer to have a separate messages for "headless" and "desktop capable". I can merge these, if you insist.

Comment (alexeysemenyukoracle) [reply test/jdk/tools/jpackage/apps/image/Hello.java:166]: Duplicated messages are not much help in understanding control flow. I'd change the new message or merge conditions. Whatever you prefer.

Comment (sashamatveev) [reply test/jdk/tools/jpackage/apps/image/Hello.java:166]: It is fine with me to have two separate messages for display and desktop, but it is better to move this check to top of function after if (GraphicsEnvironment.isHeadless()). No point to execute other code.

Comment (shipilev) [reply test/jdk/tools/jpackage/apps/image/Hello.java]: Okay, I moved the check upwards. The trace message is left intact. Note it says "desktop", there, so it is not a duplicate message.

**Review by alexeysemenyukoracle — APPROVED** [review]

**Pull request closed without merging**